BASE/src/ServerTelemetryChannel/Implementation/PartialSuccessTransmissionPolicy.cs

```diff
@@ -48,17 +49,15 @@ private static string ParsePartialSuccessResponse(Transmission initialTransmissi
             string newTransmissions = null;
             if (backendResponse.ItemsAccepted != backendResponse.ItemsReceived)
             {
-                string[] items = JsonSerializer
-                    .Deserialize(initialTransmission.Content)
-                    .Split(new[] { Environment.NewLine }, StringSplitOptions.RemoveEmptyEntries);
+                List<string> items = new List<string>(JsonSerializer.DeserializeToStrings(initialTransmission.Content));
```

Comment:
I might be missing something, this doesn't seem to improve any perf or reducing the allocation - as a list is always created.
Creating an enumerator and then convert it to a list might actually get it worse since we're paying additional cost of the enumerator.
Is there a test result to validate what do we actually get?

Reply:
We may be able to avoid deserialization here. TelemetryItems are part of `TransmissionProcessedEventArgs` - `args.Transmission.TelemetryItems`.  I had a sample built for ingestion status, there I do something like below. 

```
List<ITelemetry> failedTransmissions = new List<ITelemetry>();

foreach (var error in backendResponse.Errors)
{
failedTransmissions.Add(transmission.TelemetryItems.ElementAt(error.Index));
}
```